File: tests/benchmarks/aot_c90/containers/array_u8_slice_scan.c
#include <stdint.h>
#include <stdio.h>

#define ARRAY_U8_SLICE_SCAN_N 200000

static int64_t run(int64_t n) {
    uint8_t bytes[ARRAY_U8_SLICE_SCAN_N];
    int64_t i = 0;
    while (i < n) {
        bytes[i] = (uint8_t) i;
        i = i + 1;
    }

    uint8_t *mid = bytes + 17;
    int64_t len = n - 34;
    uint8_t sum = 0;
    i = 0;
    while (i < len) {
        sum = (uint8_t) (sum + mid[i]);
        i = i + 1;
    }
    return (int64_t) sum;
}

int main(void) {
    printf("%lld\n", (long long) run(ARRAY_U8_SLICE_SCAN_N));
    return 0;
}

File: tests/benchmarks/aot_c90/containers/array_u8_slice_scan.rs
// Rust reference for array_u8_slice_scan.xr (statement-level isomorphic, safe std-only).
// u8 accumulator wraps each round, matching Xray's sub-width wrap semantics.

fn run(n: i64) -> i64 {
    var mut bytes: Vec<u8> = Vec::new();
    var mut i: i64 = 0;
    while i < n {
        bytes.push(i as u8);
        i += 1;
    }

    var mid = &bytes[17..(n - 17) as usize];
    var mut sum: u8 = 0;
    var mut j: usize = 0;
    while j < mid.len() {
        sum = sum.wrapping_add(mid[j]);
        j += 1;
    }
    sum as i64
}

fn main() {
    println!("{}", run(200000));
}
